lib/checker.ml

```diff
@@ -476,7 +468,7 @@ and saturate_type (env: Checker_env.t) (typ: coq_P4Type) : coq_P4Type =
     TypAction (saturate_params env data_params,
                saturate_params env ctrl_params)
   | TypConstructor (type_params, wildcard_params, params, ret) ->
-    let env = Checker_env.insert_type_vars ~shadow:true type_params env in
```

Comment:
I believe I removed `~shadow:true` from `Checker_env.insert_type_var` & `Checker_env.insert_type_vars` because their underlying implementation uses `Env.insert_bare` which doesn't have a `~shadow` parameter, but perhaps this is still needed in `Checker_env.insert_type_var` & `Checker_env.insert_type_vars`?
Maybe those functions need to still perform [this check](https://github.com/verified-network-toolchain/petr4/blob/21d520543876201fcdbf5c4c207770232b35cbf7/lib/env.ml#L159) as in `Env.insert`?

Reply:
Ah, yeah, those functions need to have a shadow flag added.

Reply:
I "fixed" it. We still need to support names for types in the frontend--there are tests that do things like declare a variable of type `.Version` when there is a struct named `Version` in the top level scope.

Reply:
I put "fixed" in scare quotes here because the shadow flag defaults to true. Making it default to false causes a bunch of tests to fail, probably because there's shadowing happening that I don't understand the conditions on.

Reply:
So the surface.ml syntax should remain as it was with qualified names for type variables, and when checker.ml generates p4light syntax type variables are unqualified strings?
In that case I can probably undo the changes I made to surface ml(i), parser.mly, and checker_env.ml.

Reply:
Yes, I think that's how it should work.